### PR TITLE
feat: Add fine tuning `fit()` for Palm2TextGenerator

### PR DESCRIPTION
**Pull request by ashleyxuu**
An internal user encountered this missing overload

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #334668088🦕

### CODE REVIEW

File: tests/system/load/test_llm.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Google LLC
```
Comment (TrevorBergeron) [inline tests/system/load/test_llm.py:1]: 2024

File: bigframes/ml/llm.py [hunk 2]
```diff
@@ -132,12 +158,70 @@ def _from_bq(
...
+                DataFrame of shape (n_samples, n_features). Training data.
+            y (bigframes.dataframe.DataFrame or bigframes.series.Series:
+                Training labels.
+
+        Returns:
+            PaLM2TextGenerator: Fitted Estimator.
+        """
+        X, y = utils.convert_to_dataframe(X, y)
+
+        # TODO(ashleyxu): options= self._bqml_options
```
Comment (TrevorBergeron) [inline bigframes/ml/llm.py:212]: isn't this the very next line?

File: bigframes/ml/llm.py [hunk 3]
```diff
@@ -132,12 +158,70 @@ def _from_bq(
...
+        options = self._bqml_options
+        options["endpoint"] = self.model_name + "@001"
```
Comment (TrevorBergeron) [inline bigframes/ml/llm.py:214]: Is it always suffixed "@001"

File: bigframes/ml/sql.py [hunk 4]
```diff
@@ -177,6 +177,24 @@ def create_model(
         parts.append(f"AS {source_sql}")
         return "\n".join(parts)
 
+    # Model create and alter
```
[the remark this answers is not in the document]
Comment (GarrettWu) [reply bigframes/ml/sql.py:180]: Looks like copied from line 161. Maybe outdated. Can just remove those comments.

**Review by TrevorBergeron — APPROVED** [review]

File: bigframes/ml/sql.py [hunk 5]
```diff
@@ -177,6 +177,24 @@ def create_model(
...
+    def create_llm_remote_model(
```
Comment (GarrettWu) [inline bigframes/ml/sql.py:181]: Looks like we can refactor to merge the create functions. Not necessarily in this PR though.

File: tests/system/conftest.py [hunk 6]
```diff
@@ -537,6 +537,33 @@ def penguins_df_default_index(
     return session.read_gbq(penguins_table_id)
 
 
+@pytest.fixture(scope="session")
+def llm_fine_tune_df_default_index(
+    session: bigframes.Session,
+) -> bigframes.dataframe.DataFrame:
+    sql = """
```
Comment (GarrettWu) [inline tests/system/conftest.py:544]: You can just read_pandas to get bigframes DF.

If not used by other tests, no need to put in conftest.py. Just put in test files. Our conftest is already too large.

File: tests/system/load/test_llm.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    assert model is not None
+
+    df = model.predict(llm_remote_text_pandas_df).to_pandas()
+    assert df.shape == (3, 4)
+    assert "ml_generate_text_llm_result" in df.columns
+    series = df["ml_generate_text_llm_result"]
+    assert all(series.str.len() == 1)
+
+    # TODO(ashleyxu): After bqml rolled out version control: save, load, check parameters to ensure configuration was kept
```
Comment (GarrettWu) [inline tests/system/load/test_llm.py:38]: What is the current behavior? Is there a BQML tracking bug to put in the comment?

File: bigframes/ml/llm.py [hunk 8]
```diff
@@ -70,9 +86,19 @@ def __init__(
         model_name: Literal["text-bison", "text-bison-32k"] = "text-bison",
         session: Optional[bigframes.Session] = None,
         connection_name: Optional[str] = None,
+        max_iterations: int = 300,
+        evaluation_task: Literal[
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:90]: Let's not to expose evaluation_task. We don't auto split and don't want to run auto evaluation. 

User needs to call score() to run evaluation.

File: bigframes/ml/llm.py (hunk 3 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py:214]: BQML is at the interim with the text-bison@001 model now. text-bison@001 is retiring in July 2024 and then everything will be migrated to Gemini model by then. We're sync launch with them, so align with @001 for now, and will change to the latest gemini accordingly.

File: bigframes/ml/sql.py (hunk 5 above)
Comment (ashleyxuu) [reply bigframes/ml/sql.py:181]: sg!

File: tests/system/conftest.py (hunk 6 above)
Comment (ashleyxuu) [reply tests/system/conftest.py:544]: Done.

File: tests/system/load/test_llm.py (hunk 7 above)
Comment (ashleyxuu) [reply tests/system/load/test_llm.py:38]: filed internal bug #335492787

File: bigframes/ml/llm.py (hunk 2 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py:212]: Fixed.

File: bigframes/ml/llm.py [hunk 9]
```diff
@@ -132,12 +146,66 @@ def _from_bq(
         model_connection = model._properties["remoteModelInfo"]["connection"]
         model_endpoint = bqml_endpoint.split("/")[-1]
 
+        # Get the optional params
+        kwargs: dict = {}
+        last_fitting = model.training_runs[-1]["trainingOptions"]
+
+        dummy_arima = cls()
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:153]: not ARIMA

File: tests/system/conftest.py (hunk 6 above)
Comment (GarrettWu) [reply tests/system/conftest.py]: ?

File: bigframes/ml/llm.py [hunk 10]
```diff
@@ -51,6 +55,12 @@
 class PaLM2TextGenerator(base.BaseEstimator):
     """PaLM2 text generator LLM model.
 
+    .. note::
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:58]: Pre-GA can be put in fit() only

File: bigframes/ml/llm.py (hunk 9 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py]: Done.

File: bigframes/ml/llm.py (hunk 10 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py]: Done.

**Pull request merged**